k-distribution/include/builtin/domains.k

```diff
@@ -158,7 +158,32 @@ module MAP
   syntax KItem ::= choice(Map)                      [function, hook(MAP.choice), klabel(Map:choice)]
 endmodule
 
-module MAP-SYMBOLIC [symbolic]
+module MAP-KORE-SYMBOLIC [kore,symbolic]
+  imports MAP
+
+  rule K:KItem in_keys(M:Map K |-> _) => true
+  rule K:KItem in_keys(_:Map) => false [owise]
+
+  rule
+    #Ceil(@M:Map (@K:KItem |-> @V:KItem))
+  =>
+    {(@K in_keys(@M)) #Equals false} #And #Ceil(@M) #And #Ceil(@K) #And #Ceil(@V)
+  [anywhere, simplification]
+
+   rule
+     #Ceil(@M:Map [@K:KItem])
+   =>
+     {(@K in_keys(@M)) #Equals true} #And #Ceil(@M) #And #Ceil(@K)
+   [anywhere, simplification]
+
+  rule
```

Comment:
Why isn't `_[_ <- _]` simply `functional`?

Reply:
You're right. `_[_ <- _]` is already functional. the rule is not necessary.